### ISSUE TITLE
Issues with EXT:solr configuration deployment

### ISSUE THREAD

**Issue by fishbone1** [TYPO3-Solr/ext-solr#3390]
I don't know which template to use for this issue because I don't know if it's a documentation issue, an error due to incompatibility or a bug in the configuration files.

I couldn't find any compatibility notes in the docs therefore I installed Solr 9.0.0. I'm not using Docker. I used the installer script shipped with Solr.

Then I tried to follow the steps described here:
[Start / Getting Started / Solr](https://docs.typo3.org/p/apache-solr-for-typo3/solr/11.5/en-us/GettingStarted/Solr.html#deployment-of-ext-solr-configuration-into-apache-solr)

First, I decided to copy the cores:
```
cp -r ${EXTENSION_ROOTPATH}/Resources/Private/Solr/cores ${SOLR_INSTALL_DIR}/server/solr
```
But no cores are available after restarting Solr.

Is there something missing in solr.xml? Or is this some kind of "convention over configuration" magic that should detect the core automatically, if it's in the proper directory?

Then I tried the other option, creating the core with the API. `ext_solr_8_0_0` is probably wrong here, so I replaced it by `ext_solr_11_5_0`:

```
curl "http://localhost:8983/solr/admin/cores?action=CREATE&name=core_de&configSet=ext_solr_11_5_0&schema=german/schema.xml&dataDir=../../data/german"
```

Solr complained that `..` is not allowed in dataDir. So I changed it to `(...)dataDir=data/german`.

This resulted in another error:

> Unable to create core [core_de] Caused by: Could not load configuration from directory /var/solr/data/configsets/ext_solr_11_5_0"

So I wonder if this line is wrong in the docs:

```
cp -r ${EXTENSION_ROOTPATH}/Resources/Private/Solr/configsets ${SOLR_INSTALL_DIR}/server/solr
```

I moved the directory to the expected location before retrying the API call:

```
sudo mv /opt/solr/server/solr/configsets /var/solr/data/configsets
sudo systemctl restart solr
curl "http://localhost:8983/solr/admin/cores?action=CREATE&name=core_de&configSet=ext_solr_11_5_0&schema=german/schema.xml&dataDir=data/german"
```

Now it shows this error:

> Could not load conf for core core_de: Can't load schema /var/solr/data/configsets/ext_solr_11_5_0/conf/german/schema.xml: Plugin init failure for [schema.xml] fieldType "string_collated": Error loading class 'solr.ICUCollationField'

**Used versions (please complete the following information):**
 - TYPO3 Version: 11.5.16
 - Browser: -
 - EXT:solr 11.5.0
 - Apache Solr version: 9.0.0
 - PHP version: 8.1.12
 - MySQL version: 8.0.30

**Comment by fishbone1** [comment]
Just now I found the version matrix. Maybe we should link that in the "Getting Started / Solr" page. I carefully looked for a restriction note here and didn't think about looking into the appendix...

So probably my issue here is that Solr 9.0.0 isn't comaptible, right?

**Comment by fishbone1** [comment]
Most of the problems remain:

* Copying the cores from extension to /opt/solr/server/solr results in no avaibale cores
* Of course, ext_solr_8... has to be replaced by ext_solr_11_5_0
* `curl "http://localhost:8983/solr/admin/cores?action=CREATE&name=core_de&configSet=ext_solr_11_5_0&schema=german/schema.xml&dataDir=../../data/german"`
  again causes error:
  > Path ../../data/german disallowed due to path traversal..
* `curl "http://localhost:8983/solr/admin/cores?action=CREATE&name=core_de&configSet=ext_solr_11_5_0&schema=german/schema.xml&dataDir=data/german"`
  leads to error
  > Could not load configuration from directory /var/solr/data/configsets/ext_solr_11_5_0"

But after moving the config sets to the var directory again (see first comment above) it seems to work.

Maybe you meant the var/solr path by `${SOLR_INSTALL_DIR}` in the docs. But I think SOLR_INSTALL_DIR would be misleading in this case. Nevertheless, I also tried to `cp` everything to var/solr which lead to other error messages... So I guess some files belong to var/data and others to /opt/solr/server/solr ...

**Comment by dmitryd** [comment]
Not an answer but may be an alternative way to solve the issue: why don't you simply use docker image for the solr server? It works really well.

https://hub.docker.com/r/typo3solr/ext-solr

**Comment by fishbone1** [comment]
Yes but then we should remove the other options from the docs if it doesn't work anyway

**Comment by dkd-kaehm** [comment]
IMHO, the CURL to create the cores does not work,
but the install script should work, so the cores are created after a few moments after start.

@fishbone1: That script is deprecated and not used anymore in tests, but nevertheless we'll check that script soon.

**Comment by fishbone1** [comment]
@dkd-kaehm Which install script do you mean? I referred to the install script shipped with Solr, as described under ["Known Installers"](https://docs.typo3.org/p/apache-solr-for-typo3/solr/main/en-us/GettingStarted/Solr.html#known-installers).

I did not refer to the (not recommended) [script that shipped with the extension](https://docs.typo3.org/p/apache-solr-for-typo3/solr/main/en-us/GettingStarted/Solr.html#shipped-install-script-not-recommended)

The alternative option for curl was not a script but to copy the cores folders.
